driver/dm_multimarginloss.cpp

```diff
@@ -0,0 +1,40 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include "multimarginloss_driver.hpp"
+#include "registry_driver_maker.hpp"
+
+static Driver* makeDriver(const std::string& base_arg)
+{
+    if(base_arg == "multimarginloss")
+        return new MultiMarginLossDriver<float, double>();
+    if(base_arg == "multimarginlossfp16")
+        return new MultiMarginLossDriver<float16, double>();
+    if(base_arg == "multimarginlossbfp16")
+        return new MultiMarginLossDriver<bfloat16, double>();
```

Comment:
In other drivers, the second type provided to driver is float. Is there any reason that you provide double here ?

Reply:
yes bro double give more correct result for CPU because this op is very sensitive to fp precision issue

Reply:
Your problem with fp precision can be resolve by using `double` when calculating in `mloMultiMarginLossUnreducedForwardRunHost` and `mloMultiMarginLossReducedForwardRunHost`.
`Tcheck` should be `float` as same as it is in other operations and it should only be used for saving sequential calculation result to make sure `error = 0` in case `Tgpu = float`.

Reply:
fixed it